cmd/fleet/serve.go

```diff
@@ -809,14 +809,18 @@ the way that the Fleet server works.
 			}
 
 			if config.Prometheus.BasicAuth.Username != "" && config.Prometheus.BasicAuth.Password != "" {
-				metricsHandler := basicAuthHandler(
+				rootMux.Handle("/metrics", basicAuthHandler(
 					config.Prometheus.BasicAuth.Username,
 					config.Prometheus.BasicAuth.Password,
 					service.PrometheusMetricsHandler("metrics", promhttp.Handler()),
-				)
-				rootMux.Handle("/metrics", metricsHandler)
+				))
 			} else {
-				level.Info(logger).Log("msg", "metrics endpoint disabled (http basic auth credentials not set)")
+				if config.Prometheus.BasicAuth.Disable {
```

Comment:
I am assuming that `Disable` from if `config.Prometheus.BasicAuth.Disable ` should actually be called `isDisabled`

Reply:
If so, why do we have this convention?

Reply:
Correct. It's the convention we have in almost all `Enable/Disable` types of configs. ("almost all" because I can see a few occurrences of `Enabled` too.)

Reply:
I wonder if this is something customers use as part of the API and how complicated it will be to fix that (Not as part of this PR though)


![image](https://user-images.githubusercontent.com/121527325/226931384-72cf49ac-0230-4dea-83d3-5507c4f591a5.png)

Reply:
What I think is that maybe it's not wrong.
If the flag means an operation then it's proper to say Disable. e.g. the customer put true to `disable` something.
And I think in this case it's ok and my initial comment is wrong.

In another case where WE tell the customer/developer that something is disabled, then it should be `Disabled` as I thought in my original comment (Mistakenly)